File: packages/core/frontend/src/types/resources/daemonset.ts
import type { ManagedField } from "./shared";

export interface DaemonSet {
  Name: string;
  Namespace: string;
  Pods: string;
  NodeSelector: Record<string, string>;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  Selector: Record<string, string>;
  Images: string[];
  StrategyType: string;
  Tolerations: number;
  PodStatus: string;
}

export interface DaemonSetSummary {
  Running: number;
  Pending: number;
}

File: packages/core/kube/dto/daemonset.go
package dto

type DaemonSetRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type DaemonSet struct {
	Name          string
	Namespace     string
	Pods          string
	NodeSelector  string
	Age           string
	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	Selector      string
	Images        []string
	StrategyType  string
	Tolerations   int
	PodStatus     string
}

type DaemonSetSummary struct {
	Running int
	Pending int
}
